studio/pages/project/[ref]/reports/database.tsx

```diff
@@ -285,8 +135,127 @@ const DatabaseUsage = observer(() => {
+                          Size
+                        </Table.th>,
+                      ]}
+                      body={props.data?.map((object) => {
+                        const percentage = (
+                          ((object.table_size as number) / databaseSizeBytes) *
+                          100
+                        ).toFixed(2)
+
+                        return (
+                          <Table.tr key={`${object.schema_name}.${object.relname}`}>
+                            <Table.td>
+                              {object.schema_name}.{object.relname}
+                            </Table.td>
+                            <Table.td>
+                              {formatBytes(object.table_size)} ({percentage}%)
+                            </Table.td>
+                          </Table.tr>
+                        )
+                      })}
+                    />
+                  )}
+                </div>
+              )
+            }}
+            append={() => (
+              <div className="mt-2">
+                <Alert_Shadcn_ variant="default" className="mt-4">
+                  <AlertDescription_Shadcn_>
+                    <p>
+                      New Supabase projects have a database size of ~40-60mb. This space includes
+                      pre-installed extensions, schemas, and default Postgres data. Additional
+                      database size is used when installing extensions, even if those extensions are
+                      inactive.
+                    </p>
+
+                    <p>
+                      Please see our{' '}
+                      <a
+                        target="_blank"
+                        rel="noreferrer"
+                        className="text-brand"
+                        href="https://supabase.com/docs/guides/platform/database-size#database-space-management"
+                      >
+                        docs
+                      </a>{' '}
+                      for further information about database space and how to reduce database space.
+                    </p>
+                  </AlertDescription_Shadcn_>
+                </Alert_Shadcn_>
+              </div>
+            )}
+          />
         </section>
       </div>
     </>
   )
 })
+
+const useDatabaseReport = () => {
+  const { ref: projectRef } = useParams()
+
+  const queryHooks = queriesFactory<keyof typeof PRESET_CONFIG.database.queries>(
+    PRESET_CONFIG.database.queries,
+    projectRef ?? 'default'
+  )
+  const largeObjects = queryHooks.largeObjects()
+  const activeHooks = [largeObjects]
+
+  const handleRefresh = async () => {
+    activeHooks.forEach((hook) => hook.runQuery())
+  }
+  const handleSetParams = (params: Partial<BaseReportParams>) => {
+    activeHooks.forEach((hook) => {
+      hook.setParams?.((prev) => ({ ...prev, ...params }))
+    })
+  }
+  useEffect(() => {
+    if (largeObjects.changeQuery) {
```

Comment:
i think this can be removed as the sql should be resolved within the hook and there wouldn't be any changes to the sql after initial load.